syntax = "proto3";

// Arquivo .proto para definição do serviço gRPC TerminalService
// Autor: Guilherme Almeida Lopes
// Data de criação: 28/06/2025

package terminal;

// Serviço gRPC responsável por receber comandos do cliente e retornar suas respostas.
// Pode ser usado, por exemplo, para executar comandos remotamente.
service TerminalService {
  // Executa um comando recebido com argumentos e retorna a saída padrão, erro e código de saída.
  rpc ExecutarComando (ComandoRequest) returns (ComandoResponse);
}

// Mensagem usada pelo cliente para solicitar a execução de um comando.
message ComandoRequest {
  string comando = 1;                  // Comando a ser executado, por exemplo: "ls", "cat", etc.
  repeated string argumentos = 2;      // Lista de argumentos para o comando, ex: ["-la", "/home/user"]
  string hash_cliente = 3;            // Identificador (hash) da sessão/conexão do cliente
}

// Mensagem usada pelo servidor para retornar o resultado da execução do comando.
message ComandoResponse {
  repeated string saida = 1;          // Saída padrão (stdout), pode conter várias linhas
  string erro = 2;                    // Saída de erro (stderr), caso ocorra algum problema
  int32 codigo_saida = 3;             // Código de saída do processo (0 para sucesso, outros para falha)
}
